operator_service/routes.py

```diff
@@ -142,15 +142,20 @@ def start_compute_job():
         if _attr not in stages[0]:
             logging.error(f'Missing algorithm in stage')
             return jsonify(error=f'Missing {_attr} in stage 0'), 400
-
-    # check timeouts
+    # loop through stages and add resources
     timeout = int(os.getenv("ALGOPODTIMEOUT", 0))
-    if timeout > 0:
-        if 'maxtime' in stages[0]['compute']:
-            maxtime = int(stages[0]['compute']['maxtime'])
-            if timeout < maxtime:
-                stages[0]['compute']['maxtime'] = timeout
+    for count, astage in enumerate(workflow['stages']):
+    # check timeouts
+        if timeout > 0:
+            if 'maxtime' in workflow['stages'][count]['compute']:
```

Comment:
`workflow['stages'][count]` is the same as `astage` which is already in the loop, just use `astage`